extension/opampextension/opamp_agent.go

```diff
@@ -154,6 +162,14 @@ func (o *opampAgent) Register(capability string, opts ...opampcustommessages.Cus
 	return o.customCapabilityRegistry.Register(capability, opts...)
 }
 
+func (o *opampAgent) Ready() error {
+	return o.setHealth(&protobufs.ComponentHealth{Healthy: true})
```

Comment:
The docs for the `PipelineWatcher` interface are a little lacking; we should avoid throwing an error here or we will stop the Collector.

Reply:
good catch, was not aware of that - I have adapted the function to always return `nil` and just log potential errors in that case